src/test/mochitest/browser_dbg-xhr-breakpoints.js

```diff
@@ -4,44 +4,76 @@
 
 // Tests that a basic XHR breakpoint works for get and POST is ignored
 add_task(async function() {
-    const dbg = await initDebugger("doc-xhr.html");
-    await waitForSources(dbg, "fetch.js");
-    await dbg.actions.setXHRBreakpoint("doc", "GET");
-    invokeInTab("main", "doc-xhr.html");
-    await waitForPaused(dbg);
-    assertPausedLocation(dbg);
-    resume(dbg);
-
-    await dbg.actions.removeXHRBreakpoint(0);
-    invokeInTab("main", "doc-xhr.html");
-    assertNotPaused(dbg);
-
-    await dbg.actions.setXHRBreakpoint("doc-xhr.html", "POST");
-    invokeInTab("main", "doc");
-    assertNotPaused(dbg);
+  const dbg = await initDebugger("doc-xhr.html");
+  await waitForSources(dbg, "fetch.js");
+  await dbg.actions.setXHRBreakpoint("doc", "GET");
+  invokeInTab("main", "doc-xhr.html");
+  await waitForPaused(dbg);
+  assertPausedLocation(dbg);
+  resume(dbg);
+
+  await dbg.actions.removeXHRBreakpoint(0);
+  invokeInTab("main", "doc-xhr.html");
+  assertNotPaused(dbg);
+
+  await dbg.actions.setXHRBreakpoint("doc-xhr.html", "POST");
+  invokeInTab("main", "doc");
+  assertNotPaused(dbg);
 });
 
 // Tests the "pause on any URL" checkbox works properly
 add_task(async function() {
-    const dbg = await initDebugger("doc-xhr.html");
-    await waitForSources(dbg, "fetch.js");
-
-    // Enable pause on any URL
-    await dbg.actions.togglePauseOnAny();
-    invokeInTab("main", "doc-xhr.html");
-    await waitForPaused(dbg);
-    await resume(dbg);
-
-    invokeInTab("main", "fetch.js");
-    await waitForPaused(dbg);
-    await resume(dbg);
-
-    invokeInTab("main", "README.md");
-    await waitForPaused(dbg);
-    await resume(dbg);
-
-    // Disable pause on any URL
-    await dbg.actions.togglePauseOnAny();
-    invokeInTab("main", "README.md");
-    assertNotPaused(dbg);
+  const dbg = await initDebugger("doc-xhr.html");
+  await waitForSources(dbg, "fetch.js");
+
+  // Enable pause on any URL
+  await dbg.actions.togglePauseOnAny();
+  invokeInTab("main", "doc-xhr.html");
+  await waitForPaused(dbg);
+  await resume(dbg);
+
+  invokeInTab("main", "fetch.js");
+  await waitForPaused(dbg);
+  await resume(dbg);
+
+  invokeInTab("main", "README.md");
+  await waitForPaused(dbg);
+  await resume(dbg);
+
+  // Disable pause on any URL
+  await dbg.actions.togglePauseOnAny();
+  invokeInTab("main", "README.md");
+  assertNotPaused(dbg);
+
+  // Turn off the checkbox
+  await dbg.actions.togglePauseOnAny();
 });
+
+// Tests removal works properly
+add_task(async function() {
+  const dbg = await initDebugger("doc-xhr.html");
+
+  await Promise.all([
+    dbg.actions.togglePauseOnAny(),
+    dbg.actions.setXHRBreakpoint("1", "GET"),
+    dbg.actions.setXHRBreakpoint("2", "GET"),
+    dbg.actions.setXHRBreakpoint("3", "GET"),
+    dbg.actions.setXHRBreakpoint("4", "GET"),
+  ]);
```

Comment:
given that we're testing XHR breakpoints pane here, i think it would nice to use the helpers for clicking and typing.

We type in the watch expressions tests as well. Our rule of thumb is test the UI in the few dedicated tests and then use redux actions outside of those tests